src/main/kotlin/com/autonomousapps/tasks/ComputeAdviceTask.kt

```diff
@@ -238,43 +240,50 @@ internal class DependencyAdviceBuilder(
   val bundledTraces = mutableSetOf<BundleTrace>()
 
   init {
-    advice = computeDependencyAdvice()
+    advice = computeDependencyAdvice(projectPath)
       .plus(computeAnnotationProcessorAdvice())
       .toSortedSet()
   }
 
-  private fun computeDependencyAdvice(): Sequence<Advice> {
+  private fun computeDependencyAdvice(projectPath: String): Sequence<Advice> {
     val declarations = declarations.filterToSet { Configurations.isForRegularDependency(it.configurationName) }
     return dependencyUsages.asSequence()
       .flatMap { (coordinates, usages) ->
-        StandardTransform(coordinates, declarations, supportedSourceSets).reduce(usages)
+        StandardTransform(coordinates, declarations, supportedSourceSets).reduce(usages).map { it to coordinates }
       }
       // "null" removes the advice
-      .mapNotNull { advice ->
+      .mapNotNull { (advice, originalCoordinates) ->
+        // Make sure to do all operations here based on the originalCoordinates used in the graph.
+        // The 'advice.coordinates' may be reduced - e.g. contain less capabilities in the GradleVariantIdentifier.
         // TODO could improve performance by merging has... with find...
         when {
-          advice.isAdd() && bundles.hasParentInBundle(advice.coordinates) -> {
-            val parent = bundles.findParentInBundle(advice.coordinates)!!
-            bundledTraces += BundleTrace.DeclaredParent(parent = parent, child = advice.coordinates)
+          // The user cannot change this one:
+          // https://github.com/gradle/gradle/blob/d9303339298e6206182fd1f5c7e51f11e4bdff30/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaTestFixturesPlugin.java#L68
```

Comment:
Not sure if there is a better place to but this exception. I think it's too in individual to solve this with 'bundles' (as it's done in other cases).
For reference: https://github.com/gradle/gradle/blob/d9303339298e6206182fd1f5c7e51f11e4bdff30/subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaTestFixturesPlugin.java#L68

Probably was a mistake to hardcode this into the `test-fixtures-plugin` ... 😐